### PR TITLE
Support `output` option for pipeline/task `start` command

### PR DESCRIPTION
**Pull request by devholic**
# Changes

Support `output` option for pipeline/task `start` command.

Current `start` command output returns human friendly message, which can make other programs to hard to parse created resource.

This can be useful in CI environment, for example, 
after resource is created, we can use jq or yq for parsing created resource and wait until resource is terminated without using `showlog` option.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Run the code checkers with `make check`
- [x] Regenerate the manpages, docs and go formatting with `make generated`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

# Release Notes

```release-note
Support `output` option for pipeline/task `start` command which allows to parse created resource by other programs
```

### CODE REVIEW

**Comment by pradeepitm12** [comment]
@devholic Thanks for the PR.
I tried to `make check` worked fine on your code.
Could you please share the go lint version

**Comment by devholic** [comment]
I accidentally pushed other commit, so I reverted by force-push 🙇‍♂️ 

@pradeepitm12 I'm using golangci-lint v1.38.0

**Comment by devholic** [comment]
@pradeepitm12 `make check` succeeded after rebasing ⚡ 

@danielhelfand @piyush-garg Could you take a look?

**Comment by devholic** [comment]
Seems broken tests are fixed in main branch, so I rebased and checks seems ok 🙂

**Comment by devholic** [comment]
@danielhelfand @piyush-garg Could you take a look? 🙇‍♂️

File: pkg/cmd/pipeline/start.go [hunk 1]
```diff
@@ -131,11 +131,12 @@ like cat,foo,bar
 			if opt.UseParamDefaults && (opt.Last || opt.UsePipelineRun != "") {
 				return errors.New("cannot use --last or --use-pipelinerun options with --use-param-defaults option")
 			}
-			if opt.DryRun {
```
Comment (chmouel) [inline pkg/cmd/pipeline/start.go:134]: We are removing an option isnt it ? Maybe user would depend on this option and this update would break them, what do you think?

Comment (devholic) [reply pkg/cmd/pipeline/start.go:134]: Actually, we don't remove `dry-run` option and removing `opt.DryRun` block will not break current dry-run behavior. Only change I made is renaming `pipelineRunDryRun` function to `printPipelineRun` for reusing this function for non dry-run operations. ([diff](https://github.com/tektoncd/cli/pull/1326/files#diff-6d1b1e823072a56d8d9420b06b3b576d51901a18a703040a03c85fb10ae7f4e6L316-L318))

image

**Review by vdemeester — APPROVED** [review]
